controller/transfer/util.py

```diff
@@ -141,6 +145,19 @@ def check_and_convert_category_for_unknown(
         )
 
 
+def check_and_covert_nested_attributes_to_text(df: pd.DataFrame) -> pd.DataFrame:
+    for key in df.columns:
+        if [value for value in pick_sample(df, 10, key) if type(value) == dict]:
+            df[key] = df[key].apply(lambda x: json.dumps(x))
```

Comment:
out of curiosity, can we check the type using the function `isInstance(value,dict)?`